internal/web_server/web/views/components/dashboard_components/user_admin/pagination.templ

```diff
@@ -25,83 +26,84 @@ func handlePagination(page int, maxPages int) int {
-script insertHxGet(id string, qp string) {
-	let ps = new URL(document.location.toString()).searchParams;
-	
-	let p = ps.get(qp);
+func createButtonUrl(url url.URL, pageChange int, queryParamKey string) string {
+	parameterString := url.Query()
 
-	var url = window.location.href.split('?')[0] + "?";
+	param := parameterString.Get(queryParamKey)
 
-	var prevUrl = url;
-	var nextUrl = url;
+	if param == "" {
+		param = "1"
+	}
 
-	if (p == null) {
-		prevUrl += `${qp}=1`;
-		nextUrl += `${qp}=2`;
+	page, err := strconv.Atoi(param)
+	if err != nil {
+		page = 1
 	}
 
-	// Get the other query parameters and create the url query string
-	for( var pair of ps.entries() ) {
-		if (pair[0] != qp) {
-			prevUrl += `&${pair[0]}=${pair[1]}`;
-			nextUrl += `&${pair[0]}=${pair[1]}`;
-		}
-		else {
-			prevUrl += `&${qp}=${(parseInt(p) - 1)}`;
-			nextUrl += `&${qp}=${(parseInt(p) + 1)}`;
-		}
+	page += pageChange
+
+	if page < 1 {
+		page = 1
 	}
 
-	document.getElementById( `prev-button-${id}` ).setAttribute("href", prevUrl.replace("?&", "?"));
-	document.getElementById( `next-button-${id}` ).setAttribute("href", nextUrl.replace("?&", "?"));
+	parameterString.Set(queryParamKey, strconv.Itoa(page))
+
+	url.RawQuery = parameterString.Encode()
+
+	return url.String()
 }
 
 // pageNumber is the component showing the current page number and the maximum pages
 templ pageNumber(currentPage int, maxPages int) {
-	if maxPages != 1 {
-		<button class="border-cindigo bg-clightindigo border-y-2 text-indigo-300 px-3 py-1" disabled>
+	if maxPages > 1 {
+		<button class="border-cindigo border-y text-cindigo  px-3 py-1" disabled>
 			{ fmt.Sprintf("%v of %v", handlePagination(currentPage, maxPages), maxPages) }
 		</button>
 	} else {
-		<button class="border-cindigo border-y-2 text-cindigo px-3 py-1">
-			<input class="w-12 border border-clightindigo rounded-input text-center" type="text" value={ strconv.Itoa(currentPage) }/> of { strconv.Itoa(maxPages) }
+		<button class="border-cindigo border-y  text-indigo-300 px-3 py-1" disabled>
+			{ fmt.Sprintf("%v of %v", handlePagination(currentPage, maxPages), maxPages) }
 		</button>
 	}
 }
 
 // prevButton is the component for the previous button
-templ prevButton(page int, parentTableID string) {
+templ prevButton(page int, parentTableID string, url *url.URL, queryParamKey string) {
 	if page <= 1 {
 		<button
 			id={ fmt.Sprintf("prev-button-%v", parentTableID) }
-			class="border-cindigo bg-clightindigo border-2 text-indigo-300 font-bold px-3 py-1"
+			class="border-cindigo bg-gray-200 rounded-l-button border text-indigo-300 font-bold px-3 py-1"
 			disabled
 		>
 			&lt;
 		</button>
 	} else {
-		<a
+		<button
 			id={ fmt.Sprintf("prev-button-%v", parentTableID) }
-			href=""
-			hx-boost="true"
-			class="border-cindigo border-2 text-cindigo font-bold px-3 py-1 inline-block"
-		>&lt;</a>
+			hx-trigger="click"
+			hx-post={ createButtonUrl(*url, -1, queryParamKey) }
```

Comment:
Handle error. I have an example of error handling with HTMX in the `edit_quiz.templ`. Check `editTitle` in `admin_api.go`

Reply:
done